Xamarin.Forms.Platform.iOS/Renderers/NavigationRenderer.cs

```diff
@@ -317,6 +316,32 @@ protected virtual void OnElementChanged(VisualElementChangedEventArgs e)
 			var pack = CreateViewControllerForPage(page);
 			var task = GetAppearedOrDisappearedTask(page);
 
+			var toolbar = _secondaryToolbar;
+			var pageToPush = page as IPageController;
+			if (NavigationPage.GetHasNavigationBar(page))
+			{
+				// The below is done to get a proper difference to take away from the ContainerArea height
+				// in a similar way to ViewDidLayoutSubviews. Transitioning between a page without the
+				// NavBar to one with it can have different values on the Y after navigating back and forth.
+				var navBarFrame = NavigationBar.Frame;
+				nfloat trueBottom = 0;
+				if (navBarFrame.Y != 0)
+				{
+					if (navBarFrame.Y > 0)
+						trueBottom = navBarFrame.Y + navBarFrame.Height;
+					else
+						trueBottom = -(navBarFrame.Y);
+				}
+				else
+				{
+					trueBottom = navBarFrame.Height;
+				}
+				if (!toolbar.Hidden)
+					trueBottom += toolbar.Frame.Height;
+
+				pageToPush.ContainerArea = new Rectangle(0, 0, pack.View.Bounds.Width, pack.View.Bounds.Height - trueBottom);
```

Comment:
Do not set ContainerArea to TabbedPage because it uses its ContainerArea. [TabbedRenderer.cs:124](https://github.com/xamarin/Xamarin.Forms/blob/master/Xamarin.Forms.Platform.iOS/Renderers/TabbedRenderer.cs#L124)